OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -235,24 +235,38 @@ public String getEmail() {
         return activityLoginBinding.emailEt.getText().toString().trim();
     }
 
-    private static final int RESET_PASSWORD_REQUEST_CODE = 0;
-
     private void showResetPasswordDialog() {
-        startActivityForResult(ResetPasswordActivity.newIntent(getEmail()), RESET_PASSWORD_REQUEST_CODE);
+        resetPasswordAlert = ResetPasswordAlertDialogFragment.newInstance(this, getEmail());
+        resetPasswordAlert.setListener(this);
+        resetPasswordAlert.show(getSupportFragmentManager(), null);
     }
 
-    public void showEulaDialog() {
-        environment.getRouter().showWebViewDialog(this, getString(R.string.eula_file_link), getString(R.string.end_user_title));
+    @Override
+    public void onResult(boolean success, @Nullable String errorMessage) {
+        if (success) {
+            if (resetPasswordAlert!=null) {
+                resetPasswordAlert.dismiss();
+            }
+            showErrorDialog(getString(R.string.success_dialog_title_help), getString(R.string.success_dialog_message_help));
+        }
+        else if (resetPasswordAlert!=null) {
+            if (errorMessage!=null) {
+                resetPasswordAlert.showError(errorMessage);
+            }
+            else {
+                // this shouldn't happen, but just in case we are here, dismiss the dialog
+                resetPasswordAlert.dismiss();
```

Comment:
This is in a block that starts with else if (resetPasswordAlert != null) so the check is already happening